systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java

```diff
@@ -244,6 +245,66 @@ private void testDockerImagesForKafkaConnect() {
         LOGGER.info("Docker images verified");
     }
 
+    @Test
+    void testSecretsWithKafkaConnectWithTlsAuthentication() throws Exception {
+        final String userName = "user-example";
+
+        testMethodResources().tlsUser(KAFKA_CLUSTER_NAME, userName).done();
+
+        StUtils.waitForSecretReady(userName);
+
+        Secret userSecret = kubeClient().getSecret("user-example");
+        Secret clusterSecret = kubeClient().getSecret("connect-tests-cluster-ca-cert");
+
+        testMethodResources().kafkaConnect(KAFKA_CLUSTER_NAME, 1)
+                .editMetadata()
+                    .addToLabels("type", "kafka-connect")
+                .endMetadata()
+                .editSpec()
+                    .addToConfig("key.converter.schemas.enable", false)
+                    .addToConfig("value.converter.schemas.enable", false)
+                    .withNewTls()
+                        .addNewTrustedCertificate()
+                            .withSecretName(clusterSecret.getMetadata().getName())
+                            .withCertificate("ca.crt")
+                        .endTrustedCertificate()
+                    .endTls()
+                    .withBootstrapServers(KAFKA_CLUSTER_NAME + "-kafka-bootstrap:9093")
+                    .withNewKafkaConnectAuthenticationTls()
+                        .withNewCertificateAndKey()
+                            .withSecretName(userSecret.getMetadata().getName())
+                            .withCertificate("user.crt")
+                            .withKey("user.key")
+                        .endCertificateAndKey()
+                    .endKafkaConnectAuthenticationTls()
+                .endSpec()
+                .done();
+
+        testMethodResources().topic(KAFKA_CLUSTER_NAME, TEST_TOPIC_NAME).done();
+
+        String kafkaConnectPodName = kubeClient().listPods("type", "kafka-connect").get(0).getMetadata().getName();
+        String kafkaConnectLogs = kubeClient().logs(kafkaConnectPodName);
+
+        LOGGER.info("Verifying that in kafka connect logs are everything fine");
+        assertThat(kafkaConnectLogs, not(containsString("ERROR")));
+
+        String pathToConnectorSinkConfig = "../systemtest/src/test/resources/file/sink/connector.json";
+        String connectorConfig = getFileAsString(pathToConnectorSinkConfig);
+        LOGGER.info("Getting configuration of Connector Sink from {}", pathToConnectorSinkConfig);
+
+        cmdKubeClient().execInPod(kafkaConnectPodName, "/bin/bash", "-c", "curl -X POST -H \"Content-Type: application/json\" --data "
+                + "'" + connectorConfig + "'" + " http://localhost:8083/connectors");
+
+        sendMessages(kafkaConnectPodName, KAFKA_CLUSTER_NAME, kafkaConnectName(KAFKA_CLUSTER_NAME), TEST_TOPIC_NAME, 2);
```

Comment:
I think, instead of using this kind of producer, we should use producer outside of the kafka pods. We should change it in the future. Issue for that will be useful.